docs/src/components/sidenav.js

```diff
@@ -29,117 +28,114 @@ const NavItem = styled.li`
-        </SubNav>
-      </NavItem>
-
-      <NavItem>
-        <Link to="/installation">Installation</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/setup-client">Set up the client</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/sending-text-messages">Sending text messages</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/sending-email-messages">Sending email messages</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/check-available-templates">Check available templates</Link>
-      </NavItem>
-
-      <hr />
-
-      <NavItem>
-        <Link to="/this-is-mdx">Send a message</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Get message status</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Get a template</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Get received text messages</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Testing</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">API keys</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Limits</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Callbacks</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">API architecture</Link>
-      </NavItem>
-      <NavItem>
-        <Link to="/this-is-mdx">Support</Link>
-      </NavItem>
-    </NavList>
-  </NavWrapper>
-)
+const MAXDepth = 2
 
-export const dynamic = props => (
+const Contents = ({
+  container: Container = NestedNav,
+  pagePath,
+  items = [],
+  depth = 0,
+}) =>
+  items.length && depth < MAXDepth ? (
+    <Container>
+      {items.map((item, i) => (
+        <Fragment key={i}>
+          <NavLink to={`${pagePath}${item.url}`}>{item.title}</NavLink>
+          <Contents pagePath={pagePath} items={item.items} depth={depth + 1} />
+        </Fragment>
+      ))}
+    </Container>
+  ) : null
+
+export const DynamicSidenav = props => (
   <StaticQuery
     query={graphql`
-      query GetThePages {
+      {
         pages: allSitePage {
           edges {
             page: node {
               path
             }
           }
         }
+
+        mdxPages: allMdx {
+          edges {
+            page: node {
+              id
+              tableOfContents
+              fields {
+                pagePath
+              }
+              headings {
+                value
+                depth
+              }
+              frontmatter {
+                title
+              }
+            }
+          }
+        }
       }
     `}
-    render={data => (
-      <Nav>
-        {data.pages.edges.map(({ page }, i) => (
-          <NavItem key={i}>
-            <Link to={page.path}>{page.path}</Link>
-          </NavItem>
-        ))}
-      </Nav>
-    )}
+    render={data => {
+      const Option = ({ children }) => {
+        const edge = data.mdxPages.edges.find(
```

Comment:
We'll need the `core-js` find polyfill if we want IE support this code runs in the browser.

Reply:
good pickup. might need to do in following prs as this code gets mangled